unitary/alpha/sparse_vector_simulator.py

```diff
@@ -0,0 +1,121 @@
+
+    def _act_on_fallback_(self, action, qubits, allow_decompose):
+        if action.gate is cirq.X:
+            self._states ^= 1 << self.qubit_map[qubits[0]]
+        else:
+            dst_rows = np.zeros(len(self._states), dtype=int)
+            reverse_affected = np.zeros(1, dtype=object)
+            mask = ~0
+            for dst_bit, qubit in enumerate(qubits[::-1]):
+                src_bit = self.qubit_map[qubit]
+                bit = np.array((self._states >> src_bit) & 1, dtype=int)
+                dst_rows |= bit << dst_bit
+                reverse_affected = np.concatenate(
+                    (reverse_affected, reverse_affected | 1 << src_bit)
+                )
+                mask ^= 1 << src_bit
+            self._states &= mask
+            unique_unaffected, dst_cols = np.unique(self._states, return_inverse=True)
+            partitioned_state = np.zeros(
+                (1 << len(qubits), len(unique_unaffected)), np.complex128
+            )
+            partitioned_state[dst_rows, dst_cols] = self._amplitudes
+            np.matmul(cirq.unitary(action), partitioned_state, out=partitioned_state)
+            nz_rows, nz_cols = np.nonzero(abs(partitioned_state) > EPSILON)
+            self._states = reverse_affected[nz_rows] | unique_unaffected[nz_cols]
+            self._amplitudes = partitioned_state[nz_rows, nz_cols]
+        return True
+
+    def _perform_measurement(self, qubits):
+        raise NotImplementedError
+
+    # abstract method of OperationTarget
+    def sample(self, qubits, repetitions, prng):
+        probs = abs(self._amplitudes) ** 2
+        samples = prng.choice(self._states, size=repetitions, p=probs)
+        out = np.empty((repetitions, len(qubits)), dtype=np.uint8)
+        for j, q in enumerate(qubits):
+            out[:, j] = (samples >> self.qubit_map[q]) & 1
+        return out
+
+    def post_select(self, qubit, value):
+        assert value in (0, 1)
+        mask = 1 << self.qubit_map[qubit]
+        value *= mask
+        nonzero_indices = np.nonzero(self._states & mask == value)
+        self._states = self._states[nonzero_indices]
+        self._amplitudes = self._amplitudes[nonzero_indices]
+        self._amplitudes /= np.linalg.norm(self._amplitudes)
+
+
+class SparseSimulator(cirq.SimulatesIntermediateStateVector):
+    def __init__(self):
+        super().__init__(split_untangled_states=False)
+
+    # override
+    def _can_be_in_run_prefix(self, val):
+        return super()._can_be_in_run_prefix(val) or isinstance(
+            val, PostSelectOperation
+        )
+
+    # abstract method of SimulatorBase
+    def _create_partial_simulation_state(
+        self, initial_state, qubits, logs=None, classical_data=None
+    ):
+        assert initial_state == 0
+        return SparseSimulationState(qubits=qubits)
+
+    # abstract method of SimulatorBase
+    def _create_step_result(self, sim_state):
+        return SparseSimulatorStep(sim_state=sim_state)
```

Comment:
I would definitely add some unit tests.  You can see the unit tests for sparse simulator here:
https://github.com/quantumlib/Cirq/blob/master/cirq-core/cirq/sim/sparse_simulator_test.py

I suspect we do not need such extensive tests, but we should test all supported circuits.

You could also test something like this versus the cirq Simulator?
https://quantumai.google/reference/python/cirq/testing/random_circuit